src/riak_kv_qry_worker.erl

```diff
@@ -258,6 +258,28 @@ throttling_spawn_index_fsms(State) ->
     State.
 
 
+estimate_query_size(#state{n_subqueries_done = NSubqueriesDone,
+                           max_query_data    = MaxQueryData,
+                           sub_qrys          = SubQrys,
+                           result            = AggregateResult,
+                           qry = ?SQL_SELECT{'SELECT' = #riak_sel_clause_v1{calc_type = CalcType}} =
+                               OrigQry} = State)
+  when CalcType /= rows,
+       NSubqueriesDone == 1 ->
```

Comment:
Not that familiar with the code, but why are we constraining on `NSubqueriesDone == 1`?  Is that particular to aggregates?